broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/steps/StreamProcessorTransitionStep.java

```diff
@@ -144,6 +144,7 @@ private static StreamProcessor createStreamProcessor(
         .nodeId(context.getNodeId())
         .commandResponseWriter(context.getCommandResponseWriter())
         .maxCommandsInBatch(context.getBrokerCfg().getProcessing().getMaxCommandsInBatch())
+        // TODO: Move this to TypedRecordProcessorFactory
```

Comment:
:wrench: Can you explain why? Honestly I was kind of confused as to why some things are there and why some things are on the platform in general. Is it because the platform "doesn't care" about these things?